consensus.go

```diff
@@ -0,0 +1,235 @@
+			if choices[i].ChoiceID == choice.Id {
+				resp.Choices[i].ChoiceDescription = choice.Description
+				break
+			}
+		}
+	}
+
+	return resp, nil
+}
+
+func (mw *MultiWallet) agenda(agendas []chaincfg.ConsensusDeployment, agendaID string) *chaincfg.ConsensusDeployment {
+	for _, agenda := range agendas {
+		if agenda.Vote.Id == agendaID {
+			return &agenda
+		}
+	}
+
+	return nil
+}
+
+// SetVoteChoice sets a voting choice for the specified agenda using the
+// specified ticket hash controlled by a VSP.
+//
+// If a ticket hash isn't provided, the vote choice is set for all tickets
+// controlled by the set VSP.
+func (mw *MultiWallet) SetVoteChoice(walletID int, vspHost string, vspPubKey []byte, agendaID, choiceID, hash string, passphrase []byte) error {
+	wallet := mw.WalletWithID(walletID)
+	if wallet == nil {
+		return errors.New(ErrNotExist)
+	}
+	wal := wallet.Internal()
+
+	if vspHost == "" {
+		return fmt.Errorf("request requires a vspHost but no vspHost was provided")
+	}
+
+	// Test to ensure the passphrase is correct, and the voting is authorized.
+	err := wallet.UnlockWallet(passphrase)
+	if err != nil {
+		return translateError(err)
+	}
+	defer wallet.LockWallet()
+
+	var ticketHash *chainhash.Hash
+	if hash != "" {
+		hash, err := chainhash.NewHashFromStr(hash)
+		if err != nil {
+			return fmt.Errorf("inavlid hash: %w", err)
+		}
+		ticketHash = hash
+	}
+
+	choice := w.AgendaChoice{
+		AgendaID: agendaID,
+		ChoiceID: choiceID,
+	}
+
+	ctx := wallet.shutdownContext()
+	_, err = wal.SetAgendaChoices(ctx, ticketHash, choice)
+	if err != nil {
+		return err
+	}
+
+	vspClient, err := wallet.VSPClient(vspHost, vspPubKey)
+	if err != nil {
+		return err
+	}
+
+	// If ticket hash is provided, then set vote choice for the selected ticket.
+	if ticketHash != nil {
+		vspTicketInfo, err := wal.VSPTicketInfo(ctx, ticketHash)
+		if err != nil {
+			return err
+		}
+		// Register the provided ticket hash with the VSP client.
+		vspClient, err = wallet.VSPClient(vspTicketInfo.Host, vspTicketInfo.PubKey)
+		if err != nil {
+			return err
+		}
+		err = vspClient.SetVoteChoice(ctx, ticketHash, choice)
+		return err
+	}
+
+	// Ticket hash wasn't provided, therefore set vote choice for all tickets
+	// belonging to the selected wallet and controlled by the specififed VSP
+	var firstErr error
+	vspClient.ForUnspentUnexpiredTickets(ctx, func(hash *chainhash.Hash) error {
+		// Never return errors here, so all tickets are tried.
+		// The first error will be returned to the user.
+		err := vspClient.SetVoteChoice(ctx, hash, choice)
+		if err != nil && firstErr == nil {
+			firstErr = err
+		}
+		return nil
+	})
+
+	return firstErr
+}
+
+// GetAllAgendasForWallet returns all agendas through out the various
+// stake versions for the active network and this version of the software,
+// and all agendas defined by it for a selected wallet.
+func (mw *MultiWallet) GetAllAgendasForWallet(walletID int, newestFirst bool) (uint32, []*Agenda, error) {
```

Comment:
```suggestion
func (mw *MultiWallet) GetWalletAgenda(walletID int, newestFirst bool) (uint32, []*Agenda, error) {
```

Reply:
![Screenshot from 2022-02-28 11-42-17](https://user-images.githubusercontent.com/25265396/155969417-ecb1fe76-65c0-46fc-bee4-adddc93d9b8b.png)
![Screenshot from 2022-02-28 11-42-01](https://user-images.githubusercontent.com/25265396/155969420-83518c02-62f2-4638-9f9c-5c11b547dc5f.png)

this has been how our naming convention has been, so i believe `GetAllAgendasForWallet()`  is in line with that